Feature/messages/fix templates default reverting

VSpyridonos opened the pull request.

VSpyridonos committed:
prevent template refetching

- Updated event binding in `campaign-create.component.html` to use `onTemplateSelected` method.
- Introduced `hasTemplateLoaded` property in `campaign-create.component.ts` to manage template loading state.
- Added `onTemplateSelected` and `onSubmitCampaign` methods in `campaign-create.component.ts`.
- Modified `onStepperStepChanged` to check template loading before API calls.
- Added `@Input()` property `hasTemplateLoaded` in `campaign-basic-info.component.ts`.
- Cleaned up unused imports in `campaign-content.component.ts`.

VSpyridonos committed:
Merge remote-tracking branch 'origin/develop' into feature/messages/fix-templates-default-reverting

cleftheris commented on src/Indice.Features.Messages.App/src/app/features/campaigns/create/campaign-create.component.ts:
```diff
@@ -92,11 +98,12 @@ export class CampaignCreateComponent implements OnInit, AfterViewChecked {
 
   public onStepperStepChanged(event: StepSelectedEvent) {
-    if (event.selectedIndex === 1) {
+    if (event.selectedIndex === 1 && !this.hasTemplateLoaded) {
```
simplify if statement. 
1. `event.selectedIndex === 1` I am on the second wizard screen that contains the content views
2. `!this.hasTemplateLoaded === 1` A template id is selected and should be loaded but this has not happened yet.
3.  `if (this.templateId)` The template id is populated so I can make an api call to get the actual contents.

cleftheris replied:
The above must be as follows. Three booleans that show clear intent.
1. `isInContentStep`
2. `hasSelectedTemplate`
3. `templateNotLoaded`

VSpyridonos committed:
refactor onStepperStepChanged if condition

Improved readability by replacing nested if statements with boolean variables: isInContentStep, templateNotLoaded, and hasSelectedTemplate. This simplifies the condition for loading a template into a single check. Added non-null assertion for templateId to ensure it is defined during the API call.

cleftheris reviewed (approved):
ok
